internal/corazawaf/transaction.go

```diff
@@ -752,6 +752,51 @@ func (tx *Transaction) RequestBodyWriter() io.Writer {
 	return tx.RequestBodyBuffer
 }
 
+type Lenger interface {
+	Len() int
+}
+
+// ProcessRequestBodySize sets the maximum number of bytes that can be read from the
+// request body.
+func (tx *Transaction) AppendRequestBody(r io.Reader) (*types.Interruption, int, error) {
+	if tx.RuleEngine == types.RuleEngineOff {
+		return nil, 0, nil
+	}
+
+	if l, ok := r.(Lenger); ok {
```

Comment:
This special case doesn't seem needed with the below logic

Reply:
Practically yes, but programmatically in here we avoid the copying if an interruption is about to be triggered.